### DISCUSSION TITLE
Update garage entity names to be consistent.

### DISCUSSION THREAD

**Discussion by veleek** [home-assistant/architecture#951] [Entity Models] [answered]
An issue ([#91131](https://github.com/home-assistant/core/issues/91131)) was opened identifying the inconsistent naming of garage entities.  In some platforms they're called `garage` in others it's `garage_door`.  

I created a PR with a potential fix ([#92561](https://github.com/home-assistant/core/pull/92561)), but there are likely some wider ranging side-effects that I'm not confident I'm aware of and it's probably worth getting some feedback from others more knowledgeable with the entire backend.

As per @bdraco, I'm creating a discussion thread to try and align before moving forward with any additional work.

**Comment by jeeftor** [comment]
Are you saying that various garage integrations (cover) don't use a common name for the garage door? And you are proposing adding a constant or something?

**Reply by veleek** [reply]
That is correct, from the issue:
> For binary_sensor one can set the device_class to garage_door, for template covers the same class of entity is called garage (without the _door).

The problem with this is mostly that the generated entities will be stuff like `binary_sensor.left_garage_door` vs `cover.left_garage` for example.

See the linked PR for full details on the suggested change.  It's a little bit more involved than just adding a constant unfortuanately.

**Comment by frenck** [comment]
I don't see how we can make this change without being breaking.

That said, the device class is an internal identifier, what is the problem with keeping those as is right now? What would be the added benefit to justify the breaking change?

**Reply by veleek** [reply]
That's something I would look to you or one of the other maintainers to answer.  I'm not the creator of the original issue, so I don't have as much context into the full impact of the problem.  I was purely looking at this issue as a relatively small change that hadn't received any comments or feedback.  I wrote the PR with the intent of browsing through the code a bit more and understanding how things in the system work.  

If the final result is that this change is probably not worth it, then I think that's totally reasonable, and we can/should just close the issue and PR (with appropriate comments).

**Comment by frenck** [comment, marked as answer]
I'm going to decline this proposal. While I agree the naming is inconsistent, it does add an unneeded breaking change for something that isn't user-facing (as we translate the values in the UI).

It is breaking outside of our reach as well, as it currently can be used in things like Blueprints.

../Frenck
